src/Service/EventFinder.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Address;
+use App\Entity\Image;
+use App\Entity\Location;
+use App\Entity\Tag;
+use App\Exception\NotSupportedEntityException;
+
+final class EventFinder implements EventFinderInterface
+{
+    public function findEvents(object $entity): iterable
+    {
+        switch (get_class($entity)) {
+            case Image::class:
+                yield $entity->getEvent();
+                break;
+
+            case Tag::class:
+                yield from $this->findEventsFromTag($entity);
```

Comment:
Why don't you just

```suggestion
                yield from $entity->getEvents();
```

here (rather than calling a function that does that)?

Reply:
Because that would yield an doctrine collection, which ends up as an collection inside an iterator, which gave some debugging time to figure out.